### DISCUSSION TITLE
Docker Compose File YAML Version 2 vs 3

### DISCUSSION THREAD

**Discussion by viralganatratb** [BretFisher/docker-mastery-for-nodejs#81] [Q&A] [answered]
Hi Bret,

Thanks for this course it's super useful! I was also watching your talk: https://youtu.be/Zgx0o8QjJk4?t=1906 and see you've used in this course and in your talk recommended version 2 for Docker Compose files. However in the official Docker documentation they recommend v3:

https://docs.docker.com/compose/compose-file/compose-versioning/

Looking the the changes I can't see anything that talks about giving up local dev only concerns that you allude to in your talk. Are you able to expand on this a little more; whether this is still applicable today, especially if you're using Amazon ECS for example?

Kind regards

**Comment by BretFisher** [comment]
I've been working with them on docs, and it could likely use more effort and bringing the improved doc's idea back to life:

I started with this comment: https://github.com/moby/moby/issues/31101#issuecomment-428472115

Then this doc's PR I started: https://github.com/docker/docker.github.io/pull/7593

As well as this supplemental doc's PR: https://github.com/docker/docker.github.io/pull/7525

**Comment by BretFisher** [comment]
The practical short version is, if you never intend to use Swarm, stick with `2.4` and you'll have a superset of functionality in the `docker-compose` CLI, including the key feature of "wait-for-it" built-in, allowing compose services to wait for other services to come up *and* have successful healthchecks before they start.

You might see notes that indicate what isn't supported in 3.x branch, like this one related to above:

https://docs.docker.com/compose/compose-file/#depends_on

> Version 3 no longer supports the condition form of depends_on. The depends_on option is ignored when deploying a stack in swarm mode with a version 3 Compose file.

**Comment by viralganatratb** [comment]
Hey thanks for the speedy reply! Great thanks very much for explaining 😃

**Comment by BretFisher** [comment, marked as answer]
Update, if you're only using `docker-compose` CLI, or the new `docker compose` CLI, then versions are now deprecated and [the "version" is `Compose Spec`](https://docs.docker.com/compose/compose-file/compose-versioning/), which means you can remove the version line from your YAML all together and each feature will determine where it's supported dynamically. 

Note that Swarm is still pinned to only work with v3.x, so you'll need the version key/value in YAML if you're using it with that file.
